test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/fieldwatch/fw001.java

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package com.ibm.jvmti.tests.fieldwatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class fw001 {
+
+	public static AtomicInteger numReports = new AtomicInteger();
+	public static boolean success;
+	public static final int JNI_OK = 0;
+	public static int numExpectedReports = 5;
+	public static int numTimeouts = 10;
+
+	// Native methods to perform any initalization before running the test.
+	public native int endTest();
+	public native int startTest(String name);
+
+	// Add or remove field watches based on the information provided.
+	public static native int modifyWatches(String className, String fieldName, String fieldSignature, boolean isStatic, boolean isRead, boolean isWrite, boolean isAdd);
+
+	public static int incrementNumReports()
+	{
+		return numReports.incrementAndGet();
+	}
+
+	public static int getNumReports()
+	{
+		return numReports.get();
+	}
+
+	public static void setNumReports(int i)
+	{
+		numReports.set(i);
+	}
+
+	public String helpFieldWatch()
+	{
+		return "Fieldwatch test 1";
+	}
+
+	public boolean setup(String args)
+	{
+		setNumReports(0);
+		return startTest(this.getClass().getName().replace('.', '/')) == JNI_OK ? true : false;
```

Comment:
Similarly here - the native could just return the true/false directly.

Reply:
Fixed here: 73a2214c05a4f5c4d33c0b28846081e321ee9c65